.eslintrc.js

```diff
@@ -1,26 +1,50 @@
 module.exports = {
   root: true,
-  extends: ['@metamask/eslint-config', '@metamask/eslint-config/config/jest', '@metamask/eslint-config/config/nodejs'],
-  ignorePatterns: ['!.eslintrc.js', '!jest.config.js', 'node_modules', 'dist', 'docs', 'coverage', '*.d.ts'],
+  extends: ['@metamask/eslint-config', '@metamask/eslint-config-nodejs'],
+  ignorePatterns: [
+    '!.eslintrc.js',
+    '!jest.config.js',
+    'node_modules',
+    'dist',
+    'docs',
+    'coverage',
+    '*.d.ts',
+  ],
   overrides: [
+    {
+      files: ['*.test.ts', '*.test.js'],
+      extends: ['@metamask/eslint-config-jest'],
+      rules: {
+        'jest/no-conditional-expect': 'off',
+        'jest/no-restricted-matchers': [
+          'error',
+          {
+            resolves: 'Use `expect(await promise)` instead.',
+            // 'toBeFalsy': 'Avoid `toBeFalsy`',
+            // 'toBeTruthy': 'Avoid `toBeTruthy`',
+            toMatchSnapshot: 'Use `toMatchInlineSnapshot()` instead',
+            toThrowErrorMatchingSnapshot:
+              'Use `toThrowErrorMatchingInlineSnapshot()` instead',
+          },
+        ],
+        'jest/no-test-return-statement': 'off',
+      },
+    },
     {
       files: ['*.js'],
       parserOptions: {
-        ecmaVersion: '2018',
         sourceType: 'script',
       },
     },
     {
       files: ['*.ts'],
-      extends: ['@metamask/eslint-config/config/typescript'],
+      extends: ['@metamask/eslint-config-typescript'],
       rules: {
         // `no-shadow` has incompatibilities with TypeScript
+        // TODO: Migrate this into @metamask/eslint-config?
```

Comment:
Yes, I think we should do this.

```suggestion
        // TODO: Migrate this into @metamask/eslint-config
```